File: SlidingWindow/KConsecutiveBlackBlocks.java
package SlidingWindow;

public class KConsecutiveBlackBlocks {
    
    public static int minimumRecolors(String block, int k) {
        
        int n = block.length();
        int minOperation = Integer.MAX_VALUE;

        // looping range till substring
        for (int i = 0; i <= n - k; i++) {
            int count = 0;
            String window = block.substring(i, i+k);
            
            for (char ele : window.toCharArray()) {
                if (ele == 'W') {
                    count++;
                }
            }
            minOperation = Math.min(minOperation, count);
        }

        return minOperation;
    }

    public static void main(String[] args) {
        String block = "WBWBBBW";
        int k = 2;
        int res = minimumRecolors(block, k);
        System.out.println("Total count = " + res);
    }
}

File: SlidingWindow/KConsecutiveBlackBlocks.py
class KConsecutiveBlackBlocks:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        n = len(blocks)
        minOperations = float('inf') # this means positive infinity

        # looping through all possible windows of size k
        for i in range(n - k + 1):
            count = 0
            window = blocks[i:i + k]

            for ch in window:
                if ch == 'W':
                    count += 1

            minOperations = min(minOperations, count)

        return minOperations
    
def main():
    solver = KConsecutiveBlackBlocks()
    blocks = "WBBWWBBWBW"
    k = 7
    result = solver.minimumRecolors(blocks, k)
    print(f"Minimum recolors needed: {result}")

if __name__ == "__main__":
    main()




